### PR TITLE
feat: add apisix deb support

### PR DESCRIPTION
**Pull request by EnableAsync**
issue: https://github.com/api7/apisix-build-tools/issues/62

### CODE REVIEW

File: .github/workflows/package-apisix-deb.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                  "nodes": {
+                      "httpbin.org:80": 1
+                  }
+              }
+            }'
+          result_code=`curl -I -m 10 -o /dev/null -s -w %{http_code} http://127.0.0.1:9080/get`
+          if [[ $result_code -ne 200 ]]; then
+              printf "result_code: %s\n" "$result_code"
+              exit 125
+          fi
```
Comment (membphis) [inline .github/workflows/package-apisix-deb.yml:51]: pls add a blank line at the end of file

**Review by membphis — COMMENTED** [review]
one PR does one thing. 
you can submit a new PR to do this thing.

image

File: .github/workflows/package-apisix-dashboard.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: package apisix-dashabord
+name: Package apisix-dashabord
```
Comment (spacewander) [inline .github/workflows/package-apisix-dashboard.yml:1]: Is this change necessary for deb support?

**Review by spacewander — CHANGES_REQUESTED** [review]

File: dockerfiles/Dockerfile.apisix.deb [hunk 3]
```diff
@@ -0,0 +1,67 @@
+ARG image_base="ubuntu"
+ARG image_tag="focal"
```
Comment (spacewander) [inline dockerfiles/Dockerfile.apisix.deb:2]: Better to use the version number as the tag

File: .github/workflows/package-apisix-deb.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      BUILD_APISIX_OR_VERSION: 2.7
```
Comment (spacewander) [inline .github/workflows/package-apisix-deb.yml:13]: The name is not good. Why use OR_VERSION for apisix package?

File: test/apisix/Dockerfile.test.apisix.deb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+RUN set -x \
+    && wget https://github.com/etcd-io/etcd/releases/download/${ETCD_VERSION}/etcd-${ETCD_VERSION}-linux-amd64.tar.gz \
+    && tar -zxvf etcd-${ETCD_VERSION}-linux-amd64.tar.gz
+
+# install apisix
+RUN set -x \
+    && dpkg -i /apisix_${APISIX_VERSION}-0_amd64.deb
+
+# start etcd and test
+CMD ["sh", "-c", "(nohup etcd-$RUNNING_ETCD_VERSION-linux-amd64/etcd >/tmp/etcd.log 2>&1 &) && sleep 10 && apisix init && apisix init_etcd && /usr/local/openresty/bin/openresty -p /usr/local/apisix -g 'daemon off;'"]
```
Comment (spacewander) [inline test/apisix/Dockerfile.test.apisix.deb:32]: Why not run `apisix start` directly?

File: dockerfiles/Dockerfile.apisix.deb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+ARG iteration="0"
+ARG apisix_repo="https://github.com/apache/apisix"
+ARG checkout_v
+
+FROM ${image_base}:${image_tag}
+
+RUN set -x \
```
Comment (spacewander) [inline dockerfiles/Dockerfile.apisix.deb:9]: Better to refactor this common and turn it into a script, instead of repeating it

File: dockerfiles/Dockerfile.apisix.deb [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    && mkdir ~/.luarocks || true \
+    && luarocks config variables.OPENSSL_LIBDIR /usr/local/openresty/openssl111/lib \
+    && luarocks config variables.OPENSSL_INCDIR /usr/local/openresty/openssl111/include
+
+ARG checkout_v
+ARG iteration
+ARG apisix_repo
+ENV checkout_v=${checkout_v}
+
+RUN set -x \
```
Comment (spacewander) [inline dockerfiles/Dockerfile.apisix.deb:44]: Ditto

File: test/apisix/Dockerfile.test.apisix.deb (hunk 5 above)
Comment (EnableAsync) [reply test/apisix/Dockerfile.test.apisix.deb:25]: Because the container will exit when the process itself exits

File: .github/workflows/package-apisix-dashboard.yml (hunk 2 above)
Comment (spacewander) [reply .github/workflows/package-apisix-dashboard.yml]: Please fix other files - not just the file we pointed out

File: test/apisix/Dockerfile.test.apisix.deb (hunk 5 above)
Comment (spacewander) [reply test/apisix/Dockerfile.test.apisix.deb:25]: Could we use a placeholder like `sleep 3600s`?

File: .github/workflows/package-apisix-dashboard.yml (hunk 2 above)
Comment (spacewander) [reply .github/workflows/package-apisix-dashboard.yml]: @EnableAsync 
Please don't mark the conversation as resolved by yourself. It makes review harder. And obviously, this conversation is not resolved.

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]
